### DISCUSSION TITLE
WHDload won't work

### DISCUSSION THREAD

**Discussion by bobingabout** [BlitterStudio/amiberry#1051] [Q&A] [answered]
Okay, before I begin with the details... I very recently got my first Raspberry Pi recently, a 3A+, followed by a 4B 2GB a few days later.
The intention is to run RetroPie on both of them. So, I installed RetroPie 4.8 on both of them, different versions of it, of course, because there's a different image file for the RPi 3 than the RPi 4.
I then went and installed Amiberry using the built in system, and manually updated to version 5.5.1 on both of them, following this video as a guide. https://www.youtube.com/watch?v=81bNAS19OFs

I copied my collection of kickstarts over to the device in the BIOS folder (34500, 37300, 39106, 40063 and 40068 ones ripped from my own Amiga hardware, but also included a few others from other sources)

I then downloaded Cannon Fodder, Cannon Fodder 2, Wizkid and Yo Joe from here http://www.whdload.de/games/all.html

And, that's when the difficulties begin.

Every time I load a new game, it states that SetPatch.... not found.
And attempting to load a game either leaves me with a black screen, or red Software Failure message.

I've spent 3 days looking for and trying fixes, and nothing seems to work.

I even copied the whdboot folder over to my Windows computer, and set it up in WinUAE, with the same results.

What am I doing wrong?

The Emulator itself works, loading ROMs and RAM and setting up floppy disks and Hard drives, all works file. I saved out a couple of .uae configurations. But how do I add these to RetroPie's main screen? I tried to do it with a .sh file to run the command ``amiberry -f "conf/Yo Joe.uae" -G``, which works in that it loads that .uae file, but it always replaces the DF0 slot with the .sh file, which means they don't actually work when you're using it on a configuration of floppy disks.

Screenshots taken in WinUAE, but this is the same as what I see on Amiberry.
![Screenshot 2023-01-25 19 56 59](https://user-images.githubusercontent.com/13473775/214683685-523f8777-597a-493e-a944-652fc09054ba.png)
![Screenshot 2023-01-25 19 57 09](https://user-images.githubusercontent.com/13473775/214683687-5704f79f-4f04-42c8-9249-6f2b7bc928ef.png)
![Screenshot 2023-01-25 19 57 21](https://user-images.githubusercontent.com/13473775/214683689-718667db-8306-4f67-abcb-32b84c626d4c.png)
![Screenshot 2023-01-25 19 56 19](https://user-images.githubusercontent.com/13473775/214683693-36980766-c742-4310-970e-2b1ac3c8002b.png)

**Comment by midwan** [comment, marked as answer]
> I then downloaded Cannon Fodder, Cannon Fodder 2, Wizkid and Yo Joe from here http://www.whdload.de/games/all.html

No, you didn't download the games - That site contains only the `.slave` files for the game, not the game itself. 
The `.slave` does not work without the game files, so what you're seeing is "normal". Amiberry falls back to the default booter settings, since it cannot recognize the filename or checksum for the given game, and then WHDLoad fails to launch it because... it's not there.

The `Setpatch not found` message is not a problem, you can ignore it. It's just a warning and has nothing to do with your problem :)

You will need to source the actual game files together with the WHDLoad slave, packaged in an .LHA, for this to work.

**Reply by bobingabout** [reply]
Thank you for the help. That was the issue.

I also figured out how to do the other thing I mentioned. Just rename the first floppy disk to be whatever I want the game name to be, place it in the roms directory, and as long as it has the same name name as the .uae file, it works.

**Reply by HoraceAndTheSpider** [reply]
This will help you with a useful keyword term for searching for pre-installed games:

https://github.com/BlitterStudio/amiberry/wiki/WHDLoad-Booter-(WHDBooter)-F.A.Q.#my-whdload-package-doesnt-seem-to-be-supported-i-experience-an-error-on-load-such-as-no-slave-file-found-or-archive-contains-multiple-folders
